recipes/haproxy.rb

```diff
@@ -0,0 +1,35 @@
+#
+# Cookbook Name:: osl-jenkins
+# Recipe:: haproxy
+#
+# Copyright 2014, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+haproxy = node.default['haproxy']
+
+haproxy['enable_ssl'] = true
+haproxy['frontend_max_connections'] = '2000 \n' \
+  '  redirect scheme https if !{ ssl_fc }'
+haproxy['ssl_incoming_port'] = '443 ssl crt /etc/pki/tls/wildcard.pem ' \
+  'ciphers ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:' \
+  'ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS no-sslv3'
+haproxy['ssl_httpchk'] = false
+haproxy['members'] = [
+  {
+    'hostname' => 'jenkins',
+    'ipaddress' => '127.0.0.1',
+    'port' => '8080'
+  }
+]
```

Comment:
After switching it to the hash version, I received the following error:
```
Recipe: osl-haproxy::default
  * chef_gem[chef-rewind] action install
[2015-04-09T16:16:31+00:00] INFO: Processing chef_gem[chef-rewind] action install (osl-haproxy::default line 33)
[2015-04-09T16:16:37+00:00] INFO: chef_gem[chef-rewind] installed chef-rewind at 0.0.9

    - install version 0.0.9 of package chef-rewind


  
  ================================================================================
  Recipe Compile Error
  ================================================================================
  
  Chef::Exceptions::RecipeNotFound
  --------------------------------
  could not find recipe install_ for cookbook haproxy
  

  
  Running handlers:
[2015-04-09T16:16:37+00:00] ERROR: Running exception handlers
  Running handlers complete
[2015-04-09T16:16:37+00:00] ERROR: Exception handlers complete
[2015-04-09T16:16:37+00:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
  Chef Client failed. 1 resources updated in 16.168889148 seconds
[2015-04-09T16:16:37+00:00] ERROR: could not find recipe install_ for cookbook haproxy
[2015-04-09T16:16:37+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

As I changed nothing else, I'm confident that using that syntax won't work as expected and does not get deep merged.